### ISSUE TITLE
Session created but application size=0

### ISSUE THREAD

**Issue by cavaIV** [ios-driver/ios-driver#264]
Hey guys!

Sometimes when I start a test the ios-driver starts the simulator but crashes right after with the following error msg:
WARNING RemoteIOSWebDriver.<init> session created but application size=0

After this the ios-driver closes and starts again.

This also occurs when i try to open a few HTML5 games from my site (open game ==> go back to website, open game ==> ...). It normally occurs when i try to open the 3rd game.

When this happens ios-server.jar also throws an error:
http://i59.tinypic.com/1zg84jl.jpg

**Comment by julianhaslinger** [comment]
Can anyone help with this one, please?

**Comment by freynaud** [comment]
Is it using mobile safari or a native app ? Can you attach a test I can reproduce ?

**Comment by cavaIV** [comment]
we are using Mobile Safari

**Comment by cavaIV** [comment]
I'm pretty sorry but i can't send you one of our tests (it's company code)

Is there any certain method call / workflow that you need to know?

**Comment by freynaud** [comment]
can you show the code of the test at least, so that I get an idea of what happens ? application size = 0 means something went wrong the the webkit protocol.

**Comment by cavaIV** [comment]
IWebDriver driver = null;
            var capabilities = DesiredCapabilities.IPhone();
            capabilities.SetCapability("AcceptSslCertificates", "true");
            capabilities.SetCapability("language", "en");
            capabilities.SetCapability("rotatable", "true");

```
        try
        {
            driver = new RemoteWebDriver(new Uri("http://QAWebRegiOS:4444/wd/hub"), capabilities);
        }
        catch (Exception error)
        {
            if (error.Message.Contains("no devices available"))
            {
                RestartiOSNode();
                driver = new RemoteWebDriver(new Uri("http://QAWebRegiOS:4444/wd/hub"), capabilities);
            }
        }

        ((IJavaScriptExecutor)driver).ExecuteScript("window.location=arguments[0]", TestSuiteConfiguration.SiteUrlMobile);
        Thread.Sleep(3000);
        driver.SwitchTo().Alert().Accept();
```

That's how we initialize our webdriver.
TestSuiteConfiguration.SiteUrlMobile is just a string with the url in it

**Comment by julianhaslinger** [comment]
@freynaud, can you help us with this one? We really have no idea what to do with this right now and your help would be appreciated. Thanks!

**Comment by freynaud** [comment]
is safari loosing the focus ? normally the app size is only checked when the webkit protocol is initialised.
If you can reproduce on a publicly accessible site, it would really help with the troubleshooting.

**Comment by cavaIV** [comment]
I will try to find a public page where I can reproduce this problem.

Another strange thing i recognized is, that mobile safari pops up and normally switches to "about:blank" before switching to the given URL (via our Javascript command) but when it fails it doesn't switch to about:blank, instead it shows "Search or enter an address" and leaves an empty page.

Here's a screenshot with the error and the ios simulator:
http://imgur.com/vRzh1DA

**Comment by freynaud** [comment]
are you saying that in your test, you have a moment when safari goes back to the about:blank page ?
if you can join irc #ios-driver we could debug that a bit more real time

**Comment by cavaIV** [comment]
yes safari goes back to the about:blank page right after we start it.

When our tests work the following things happens:
-) Safari opens ==> address bar shows: 'Search or enter an address'
-) Safari switches to "about:blank"
-) Our javascript get executed ==> safari switches to our website

I already tested it with "www.google.com" instead of our website and the error also occurs.

It seems like once the error occurs it's something wrong and no more test will be executed correctly.

I already joined irc #ios-driver

**Comment by cavaIV** [comment]
It seems like the error occurs between these two lines of code:
 driver = new RemoteWebDriver(new Uri("http://QAWebRegiOS:4444/wd/hub"), capabilities);
and
((IJavaScriptExecutor)driver).ExecuteScript("window.location=arguments[0]", TestSuiteConfiguration.SiteUrlMobile);

**Comment by freynaud** [comment]
can you ping me on irc ?
what's your reason for using execute script rather than driver.get(url)

**Comment by cavaIV** [comment]
I can join irc from 8:00 to 15:30

I'm executing a script cause otherwise I can't open a https page

**Comment by cavaIV** [comment]
@freynaud 
is it possible the error occurs cause of the javascript i'm executing for the https page?

**Comment by freynaud** [comment]
it shouldn't. The application list is sent before anything else happens.
